src/common/GPUUtils.hpp

```diff
@@ -19,6 +19,19 @@
 namespace rajaperf
 {
 
+/*!
+ * \brief Routine to check whether number of args in signature of the
+ *        given kernel matches the number of items in the given tuple.
+ */
+template <typename... Formals, typename... Actuals>
+void checkArgsCount(void (*kernel)(Formals...),
+                    Actuals...)
+{
+  (void) kernel; // to prevent compiler warning
+  static_assert(sizeof...(Formals) == sizeof...(Actuals),
+                "Argument Count Mismatch");
```

Comment:
```suggestion
                "Argument Count Mismatch");
    using int_array = int[];
    int_array {[](){
        static_assert(std::is_same<std::decay_t<Formals>, std::decay_t<Actuals>>::value, "Wrong arg type");
        return 0;
        }()...};
```

Reply:
This checks that the args are also of the right type.

Reply:
That would be fairly obvious in the compiler error message I think. Also, other issues could occur such as providing args of the same type in the wrong order which is more serious.  Plus, I'm not completely clear on what you are proposing here.